vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -577,7 +580,9 @@ - (void)onGrantCaptcha:(const std::string &)image hint:(const std::string &)hint
 
 - (void)fetchGrants:(const std::string &)lang paymentId:(const std::string &)paymentId
 {
-  ledger->FetchGrants(lang, paymentId);
+  ledger->FetchGrants(lang, paymentId, ^(ledger::Result result){
+    // handle this?
```

Comment:
This is the fetchGrants method from `LedgerClient`. Perhaps the LedgerClient method should add a callback if it wants to know when it finishes like you did to the main method? If not you can just replace this comment stating that the ledger lib will handle the rest internally

Reply:
I just removed this from `LedgerClient`, I'm not sure why we had it in the first place and a comment in `ledger_client.h` said that it can be removed.

Reply:
Oh, great! You can then remove it from NativeLedgerClient.h, NativeLedgerClient.mm, NativeLedgerClientBridge.h, and then finally this whole method in BATBraveLedger.mm